### PR TITLE
feat: implement v2 migration tool with YAML-driven transformations

### PR DESCRIPTION
**Pull request by tamas-jozsa**
## Summary

  This PR introduces a new v2 migration tool that automates the migration of
  Terraform configurations between different versions of the Cloudflare provider. The
   tool uses a YAML-driven approach for most transformations, with extensibility for
  custom migrations when needed.

  ## Key Features

  ### 🚀 Core Functionality
  - **Automatic migration** from provider v4 to v5 (and beyond)
  - **YAML-driven transformations** - no code required for most migrations
  - **Token-based HCL parsing** - preserves formatting, comments, and expressions
  - **Safe by default** - built-in backup and rollback capabilities
  - **Preview mode** - see changes before applying them

  ### 🔧 Transformations Implemented
  - **Attribute operations**: Renames, removals, default values
  - **Structural changes**:
    - Convert blocks to lists and vice versa
    - Flatten nested objects with configurable depth
    - Split objects into flat attributes
    - Merge attributes into objects/lists
  - **Conditional logic**: Remove attributes based on resource state
  - **Type conversions**: Handle set→list, string→int, etc.

  ### 📚 Documentation
  - Comprehensive README with quick start guide
  - YAML configuration reference
  - Custom migration guide for complex scenarios
  - Real-world examples for all transformation types
  - Testing guidelines with unit and integration test examples

  ## Technical Implementation

  The migration tool follows a modular architecture:

  ┌─────────────┐     ┌──────────────┐     ┌─────────────┐
  │     CLI     │────▶│ Orchestrator │────▶│  Registry   │
  └─────────────┘     └──────────────┘     └─────────────┘
                              │                     │
                              ▼                     ▼
                      ┌──────────────┐     ┌─────────────┐
                      │  Migration   │◀────│ Transformers│
                      │    Engine    │     │ (HCL & JSON)│
                      └──────────────┘     └─────────────┘

  ### Key Design Decisions
  - **Token-based parsing**: Uses `hclwrite` AST manipulation to preserve HCL
  formatting
  - **YAML-first approach**: Most migrations require only YAML configuration
  - **Extensible**: Easy to add new transformations or custom migration logic
  - **Resource-specific**: Each resource type can define its own migrations
  - **Version chaining**: Automatically finds migration paths (e.g., v4→v5→v6)

  ## Testing

  All transformations include comprehensive test coverage:
  - ✅ 20/20 test suites passing
  - ✅ Unit tests for each transformer
  - ✅ Integration tests with real HCL configurations
  - ✅ Edge cases: empty separators, nil mappings, complex expressions
  - ✅ Benchmarks for performance-critical operations

  ## Usage Example

  ```yaml
  # Simple YAML migration configuration
  version: "1.0"
  resource_type: cloudflare_access_application
  migration:
    from: "4.*"
    to: "5.*"

  attribute_renames:
    domain: hostname

  blocks_to_lists:
    - destinations

  remove_fields:
    - deprecated_field

  default_values:
    type: "self_hosted"

  Files Changed

  Core Implementation

  - cmd/migrate-v2/transformations/config/ - All config transformers
  - cmd/migrate-v2/core/migration.go - Base migration framework
  - cmd/migrate-v2/README.md - Main documentation

  Documentation

  - Simplified and consolidated 7 documentation files
  - Created human-friendly guides focused on usage over implementation
  - Added real-world examples for all transformation types

  Breaking Changes

  None - this is a new tool that doesn't affect existing functionality.

  Future Work

  - Add support for more complex type conversions
  - Implement state migration for nested structures
  - Add interactive mode for migration preview
  - Support for custom validation rules
  - Performance optimizations for large configurations

  Review Checklist

  - Code follows project conventions
  - Tests pass locally
  - Documentation is complete and accurate
  - Examples work as documented
  - No hardcoded values or secrets

### CODE REVIEW

File: cmd/migrate-v2/Makefile [hunk 1]
```diff
@@ -0,0 +1,122 @@
+.PHONY: test test-unit test-integration test-verbose test-coverage clean build help
+
+# Build the migrate binary
+build:
+	@echo "Building migrate-v2 binary..."
```
Comment (mgirouard) [inline cmd/migrate-v2/Makefile:5]: You can clean up all the `@echo` stuff by replacing them with bash comments. Much cleaner code, slightly different output.

```diff
- @echo "Building migrate-v2 binary..."
+ # Building migrate-v2 binary...
```

File: cmd/migrate-v2/Makefile [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	@go build -o bin/migrate-v2 main.go
```
Comment (mgirouard) [inline cmd/migrate-v2/Makefile:6]: I don't recommend silencing any commands in make. It's always better to see what executes.

You can always call `make -s` to suppress commands.
